src/graphics/program-lib/programs/lit-shader.js

```diff
@@ -608,17 +608,31 @@ class LitShader {
 
         let code = "";
 
-        if (options.clearCoat > 0) {
-            code += '#define CLEARCOAT\n';
-
-            // enable clear-coat path in clustered chunk
-            code += "#define CLUSTER_CLEAR_COAT\n";
-        }
-
         if (options.opacityFadesSpecular === false) {
             code += 'uniform float material_alphaFade;\n';
         }
 
+        if (options.useSpecular) {
+            code += "#define LIT_SPECULAR\n";
+
+            if (this.reflections) {
+                code += "#define LIT_REFLECTIONS\n";
+            }
+
+            if (options.clearCoat) {
+                code += "#define LIT_CLEARCOAT\n";
+            }
+
+            if (options.fresnelModel > 0) {
+                code += '#define LIT_SPECULAR_FRESNEL\n';
+            }
+
+            // enable conserve energy path in clustered chunk
+            if (options.conserveEnergy) {
+                code += "#define LIT_CONSERVE_ENERGY\n";
+            }
+        }
```

Comment:
Just as a general point, when generating GLSL shaders, would it make sense to prepend all `#define`s to the start of the code? cc @slimbuck 
Note that I'm not suggesting we make this change in this PR!

Reply:
Yes agree. I'd actually suggest adding a `globalDefines` variable and adding the defines to that. Then at the end of the function where the final shader is combined insert it at the right place (before this.varyings probably).

Reply:
Sounds like a good idea, we can provide a ```addDefine('LIT_SPECULAR_FRESNEL')``` and have that be added at shader generation time. 

I can do it for this PR if you think it would fit, and then have other defines merged over to it later.

Reply:
I made this change in the PR. We now inject defines on ```_fsGetBeginCode```.